packages/labs/gen-manifest/src/index.ts

```diff
@@ -24,43 +24,38 @@ import {
 import {FileTree} from '@lit-labs/gen-utils/lib/file-utils.js';
 import type * as cem from 'custom-elements-manifest/schema';
 
-const isNotEmpty = (v: unknown) =>
-  v !== undefined &&
-  (typeof v !== 'boolean' || v === true) &&
-  (typeof v !== 'string' || v.length > 0) &&
-  (!Array.isArray(v) || v.length);
-
-const pickIfNotEmpty = <O, K extends keyof O>(model: O, name: K) => {
-  const obj: {[key in K]?: O[K]} = {};
-  const v = model[name];
-  if (isNotEmpty(v)) {
-    obj[name] = model[name];
+/**
+ * For optional entries in the manifest, if the value has no meaningful value
+ * (i.e. it's an empty string or array or `false`), return `undefined` so that
+ * we don't serialize a key with a meaningless value to JSON, to cut down size
+ * of the manifest (note that `JSON.serialize` will not emit keys with
+ * `undefined` values)
+ */
+const ifNotEmpty = <T>(v: T): T | undefined => {
+  if (
+    v === undefined ||
+    (typeof v === 'boolean' && v === false) ||
+    (typeof v === 'string' && v.length === 0) ||
+    (Array.isArray(v) && v.length === 0)
```

Comment:
This can still be simplified, ie with `===` the `typeof v === 'boolean'` isn't necessary, and the undefined check isn't needed:

```suggestion
v === false || ((typeof v === 'string' || Array.isArray(v)) && v.length === 0)
```